src/App.tsx

```diff
@@ -1,24 +1,373 @@
+                  type="checkbox"
+                  className="todo__status"
+                />
+              </label>
+              <span data-cy="TodoTitle" className="todo__title">
+                {tempTodo.title}
+              </span>
+              <button
+                type="button"
+                className="todo__remove"
+                data-cy="TodoDelete"
+                onClick={() => {
+                  deleteTodo(tempTodo.id);
+                }}
+              >
+                ×
+              </button>
+              <div
+                data-cy="TodoLoader"
+                className="modal overlay is-active"
+              >
+                <div className="modal-background has-background-white-ter" />
+                <div className="loader" />
+              </div>
+            </div>
+          )}
+        </section>
+
+        {todos.length > 0 && (
+          <footer className="todoapp__footer" data-cy="Footer">
+            <span className="todo-count" data-cy="TodosCounter">
+              {`${todos.filter(todo => !todo.completed).length} items left`}
+            </span>
+            <nav className="filter" data-cy="Filter">
+              <a
+                href="#/"
+                className={classNames('filter__link ', { selected: filterParam === 'All' })}
+                data-cy="FilterLinkAll"
+                onClick={() => setFilterParam('All')}
+              >
+                All
+              </a>
+              <a
+                href="#/active"
+                className={classNames('filter__link ', { selected: filterParam === 'Active' })}
+                data-cy="FilterLinkActive"
+                onClick={() => setFilterParam('Active')}
+              >
+                Active
+              </a>
+              <a
+                href="#/completed"
+                className={classNames('filter__link ', { selected: filterParam === 'Completed' })}
+                data-cy="FilterLinkCompleted"
+                onClick={() => setFilterParam('Completed')}
+              >
+                Completed
+              </a>
+            </nav>
+            <div>
+              {todos.filter(todo => todo.completed).length > 0 && (
+                <button
+                  type="button"
+                  className="todoapp__clear-completed"
+                  data-cy="ClearCompletedButton"
+                  onClick={() => onDeleteCompleted()}
+                >
+                  Clear completed
+                </button>
+              )}
+            </div>
+          </footer>
+        )}
+      </div>
+      <div
+        data-cy="ErrorNotification"
+        className={classNames('notification is-danger is-light has-text-weight-normal', { hidden: !errorMsg })}
+      >
+        <button
+          data-cy="HideErrorButton"
+          type="button"
+          className="delete"
+          onClick={() => (setErrorMsg(null))}
+        />
+        {errorMsg}
+      </div>
+    </div>
```

Comment:
in `ErrorMessage` should be implemented this functionality (from react_todo-app-loading-todos)
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/91946036/667a3e44-9852-42ec-9541-80441cc771ed)